pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -39,6 +39,11 @@ const ServiceAnnotationLoadBalancerInternal = "service.beta.kubernetes.io/azure-
 // to specify what subnet it is exposed on
 const ServiceAnnotationLoadBalancerInternalSubnet = "service.beta.kubernetes.io/azure-load-balancer-internal-subnet"
 
+// ServiceAnnotationLoadBalancerPublicIPAddressID is the annotation used on the service
+// to specify the resource ID of the public IP address,
+// which can be used to specify public IP address that's not in the same resource group as the cluster
+const ServiceAnnotationLoadBalancerPublicIPAddressID = "service.beta.kubernetes.io/azure-load-public-ip-id"
```

Comment:
It seems like this is used only to derive a resource group name: the actual Public IP resource is still located by looking up the `loadBalancerIP` value in the public IPs in that resource group.  The code doesn't require that the Public IP resource in the annotation is the one that matches the `loadBalancerIP` address.  Given that, perhaps we should just have the annotation be a resource group name here instead?

Reply:
Annotating with just resource group actually makes sense to me. I was carried away by the issue description 'resource id'.